libevm/VMFactory.cpp

```diff
@@ -86,7 +86,8 @@ void setVMKind(const std::string& _name)
 
     // If no match for predefined VM names, try loading it as an EVMC VM DLL.
     evmc_loader_error_code ec;
-    g_evmcDll.reset(new EVMC{evmc_load_and_create(_name.c_str(), &ec)});
+    evmc_instance *instance = evmc_load_and_create(_name.c_str(), &ec);
+    g_evmcDll.reset(nullptr);
```

Comment:
Not sure we need this.

Also could have an assertion that if `ec != EVMC_LOADER_SUCCESS` then `instance == nullptr`

Reply:
I think this is ok to release the previous VM. You can even do it before `evmc_load_and_create`. 
Also, you don't need `nullptr` for argument, it can be just `.reset()`.